website/pages/docs/k8s/upgrade/index.mdx

```diff
@@ -7,63 +7,303 @@ description: Upgrade Consul on Kubernetes
+   you may need to also upgrade your Helm chart version at the same time.
+1. Set `global.consul` in your `values.yaml` to the desired version:
+   ```yaml
+   global:
+     image: consul:1.8.3
+   ```
+1. Determine your current installed chart version
+
+```bash
+helm list -f consul
+NAME  	NAMESPACE	REVISION	UPDATED        	STATUS  	CHART        	APP VERSION
+consul	default  	2       	2020-09-30 ...	deployed	consul-0.24.0	1.8.2
+```
+
+In this example, I'm running version `0.24.0` (from `consul-0.24.0`).
+
+1. Run `helm upgrade consul hashicorp/consul --version 0.24.0 -f /path/to/my/values.yaml`.
+
+   **Before upgrading, be sure you've read the other sections on this page.**
+
+~> NOTE: It's important to always set the `--version` flag because otherwise Helm
+will use the most up-to-date version in its local cache which may result in an
+unintended upgrade.
+
+## Determining What Will Change
+
+Before upgrading, it's important to understand what changes will be made to your
+cluster. For example, you will need to take more care if your upgrade will result
+in the Consul server statefulset being redeployed.
+
+There is no built-in functionality to Helm that shows what a helm upgrade will
+change. There is however a Helm plugin [helm-diff](https://github.com/databus23/helm-diff)
+that can be used.
+
+1. Install `helm-diff` with:
+
+```bash
+helm plugin install https://github.com/databus23/helm-diff
+```
+
+1. If you are updating your `values.yaml` file, do so now.
+1. Take the same `helm upgrade` command you were planning to execute but run `helm diff upgrade` instead of `helm upgrade`:
+
+```bash
+helm diff upgrade consul hashicorp/consul --version 0.24.1 -f /path/to/your/values.yaml
+```
+
+This will print out the manifests that will be updated and their diffs.
+
+1. To see only the objects that will be updated, add `| grep "has changed"`:
+
+```bash
+helm diff upgrade consul hashicorp/consul --version 0.24.1 -f /path/to/your/values.yaml |
+  grep "has changed"
+```
+
+1. Take specific note if `consul, DaemonSet` or `consul-server, StatefulSet` are listed.
+   This means that your Consul client daemonset or Consul server statefulset (or both) will be redeployed.
+
+If either is being redeployed, we will follow the same pattern for upgrades as
+on other platforms: the servers will be redeployed one-by-one and then the
+clients will be redeployed in batches. Read [Upgrading Consul](/docs/upgrading) and then continue
+reading below.
+
+If neither the client daemonset nor the server statefulset is being redeployed
+then you can continue with the helm upgrade without any specific sequence to follow.
+
+## Service Mesh
+
+If you are using Consul's service mesh features (as opposed to the [service sync](/docs/k8s/service-sync)
+functionality) you must be aware of the behavior of the service mesh during upgrades.
+
+Consul clients run as a daemonset across all Kubernernetes nodes. During an upgrade,
```

Comment:
```suggestion
Consul clients operate as a daemonset across all Kubernernetes nodes. During an upgrade,
```

Reply:
🚫 "run" I take it!

Reply:
Yeah :) I just did it everywhere so you could just accept the suggestion instead of having to go make the edits yourself.

Reply:
Yeah I would have but I also wanted to check the formatting so it was easier to follow along and make the suggested changes locally.